src/transform/transform.ts

```diff
@@ -938,3 +939,96 @@ function parseComments(comment: string): string {
+  if (modelSchemas){
+    for (const schema of modelSchemas){
+      if (!referencedSet.has(schema)){
+        schema.language.go!.omitType = true;
+      }
+    }
+  }
+}
+
+// This function iterate all the operations in all operation groups and try to find all the referenced types (including choices, sealedChoices and objects).
+// For each operation, we will aggregate all the parameters' type of the operation and put them into the referencedTypes set.
+// Also, all responses body types and header types will be put into the referencedTypes set.
+// Some special cases: exceptions response types, x-ms-odata types.
+function iterateOperations(model: CodeModel, referencedTypes: Set<Schema>) {
+  for (const group of values(model.operationGroups)) {
+    for (const op of values(group.operations)) {
+      for (const param of values(aggregateParameters(op))) {
+        dfsSchema(param.schema, referencedTypes);
+      }
+      // We do not use error responses' definition to unmarshal response. So exceptions will be ignored.
+      // const responses = values(op.responses).concat(values(op.exceptions));
+      const responses = values(op.responses);
+      for (const resp of values(responses)) {
+        if (isSchemaResponse(resp)) {
+          dfsSchema(resp.schema, referencedTypes);
+        }
+        const httpResponse = <HttpResponse>resp.protocol.http;
+        for (const header of values(httpResponse.headers)) {
+          dfsSchema(header.schema, referencedTypes);
+        }
+      }
+      // Such odata definition is not used directly in operation. But it seems the model is a detailed definition for some string param. Reserve for now.
+      if (op.extensions?.['x-ms-odata']) {
+        const schemaParts = op.extensions['x-ms-odata'].split('/');
+        const schema = values(model.schemas.objects).where((o) => o.language.default.name === schemaParts[schemaParts.length - 1]).first();
+        if (schema) {
+          dfsSchema(schema, referencedTypes);
+        }
+      }
+    }
+  }
+}
+
+// This function will do a depth first search for the root types.
+// All visited types will be put into referencedTypes set.
+// Objects children/parents will also be searched.
+function dfsSchema(schema: Schema, referencedTypes: Set<Schema>) {
+  if (referencedTypes.has(schema)) return;
+  referencedTypes.add(schema);
+  if (isObjectSchema(schema)) {
+    const allProps = aggregateProperties(schema);
+    for (const prop of allProps){
+      dfsSchema(prop.schema, referencedTypes);
+    }
+    // If schema is a discriminator, then reserve all the children
+    if (schema.discriminator) {
+      for (const child of values(schema.children?.all)) {
+        dfsSchema(child, referencedTypes);
+      }
+    }
+    // If schema is has allOf, then reserve all the parents
+    for (const parent of values(schema.parents?.immediate)) {
+      if (isObjectSchema(parent)) {
+        dfsSchema(parent, referencedTypes);
+      }
+    }
```

Comment:
Should we preserve the parent schema or omit the parent schema? I believe the parent schemas are one of the biggest source of unused models since we are flattening the properties from its parent into itself